### ISSUE TITLE
Migrate to BaseExceptionGroup

### ISSUE THREAD

**Issue by macalinao** [pgjones/hypercorn#131]
I'm getting this annoying error every time I start quart-trio:

```
/Users/pgjones/Library/Caches/pypoetry/virtualenvs/python-template-zJCAta4K-py3.10/lib/python3.10/site-packages/hypercorn/trio/task_group.py:25: TrioDeprecationWarning: trio.MultiError is deprecated since Trio 0.22.0; use BaseExceptionGroup (on Python 3.11 and later) or exceptiongroup.BaseExceptionGroup (earlier versions) instead (https://github.com/python-trio/trio/issues/2211)
```

Would appreciate the update!

**Comment by Zac-HD** [comment]
See also https://github.com/pgjones/hypercorn/pull/126

**Comment by pgjones** [comment]
I think this is fixed in d2c5cd00cf2408ead6bfb1971554d5f1fb8675db
